pkg/projectfile/projectfile.go

```diff
@@ -965,18 +965,13 @@ func createCustom(params *CreateParams, lang language.Language) (*Project, error
 		if hostOverride := os.Getenv(constants.APIHostEnvVarName); hostOverride != "" {
 			host = hostOverride
 		}
-		u, err := url.Parse(fmt.Sprintf("https://%s/%s/%s", host, params.Owner, params.Project))
-		if err != nil {
-			return nil, errs.Wrap(err, "url parse new project url failed")
-		}
-		q := u.Query()
-
+		pjf := NewProjectField()
+		pjf.LoadProject("https://" + host)
+		pjf.SetNamespace(params.Owner, params.Project)
 		if params.BranchName != "" {
-			q.Set("branch", params.BranchName)
+			pjf.SetBranch(params.BranchName)
 		}
-
-		u.RawQuery = q.Encode()
-		params.ProjectURL = u.String()
+		params.ProjectURL = pjf.String()
```

Comment:
This is to avoid code duplication. I was using `NewProjectField()` in the buildplanner to construct the "Project" field in the checkout info block of buildscripts, and found that we can do the same thing here.